### PR TITLE
Revamp logging and validation

### LINKED ISSUE #238: Validate service account auth keys and improve error message
## Execution failed for task ':app:publishApkRelease'.

com.google.api.client.googleapis.json.GoogleJsonResponseException: 401 Unauthorized

``java        
play {          
        track = 'alpha'              
    }               
    signingConfigs {               
        debug {               
            storeFile file('app.jks')             
            keyAlias KEY_ALIAS               
            keyPassword KEY_PASSWORD                
            storePassword KEYSTORE_PASSWORD                    
        }
        release {               
             storeFile file('app.jks')                 
            keyAlias KEY_ALIAS                    
            keyPassword KEY_PASSWORD                  
            storePassword KEYSTORE_PASSWORD                  
        }                
    }
``

### LINKED ISSUE #244: Make gradle-play-publisher work with Jenkins Android Signing Plugin
## Problem

We do not let Gradle sign our release apks anymore, but instead use the "Jenkins Android Signing Plugin", see https://github.com/jenkinsci/android-signing-plugin for this purpose.

Unfortunately, gradle-play-publisher strictly depends on "signingConfig" statement in build.gradle in order to create the Gradle publish tasks correctly.

## Solution

Drop this precondition and make gradle-play-publisher work with apks that have been signed after the Gradle build is finished (like by a Jenkins plugin).

This issue relates to issue https://github.com/Triple-T/gradle-play-publisher/issues/233

### LINKED ISSUE #268: Revamp logging and validation
This issue coalesces multiple requests related to logging. We need to:
- validate signing configs
- log what's being uploaded
- show progress while uploading
- clarify error messages
- ensure important logs show up in the build without having to pass a `--info` flag

### PR DESCRIPTION
**Pull request by SUPERCILEX**
Fixes #244, fixes #238, fixes #268

### CODE REVIEW

**Comment by SUPERCILEX** [comment]
Closing to not spam you guys with my attempts to get this working.

**Comment by SUPERCILEX** [comment]
Here ya go: https://youtu.be/ld8-awABW4c

File: plugin/src/main/kotlin/com/github/triplet/gradle/play/internal/Progress.kt [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package com.github.triplet.gradle.play.internal
+
+import com.google.api.client.googleapis.media.MediaHttpUploader
+import com.google.api.services.androidpublisher.AndroidPublisherRequest
+import org.gradle.internal.logging.progress.ProgressLogger
+import kotlin.math.roundToInt
+
+internal fun <T> AndroidPublisherRequest<T>.trackProgress(
```
Comment (bhurling) [inline plugin/src/main/kotlin/com/github/triplet/gradle/play/internal/Progress.kt]: This is really nice. Let's call it trackUploadProgress as there are some very phrases in here that are very specific to artifact uploads

Comment (SUPERCILEX) [reply plugin/src/main/kotlin/com/github/triplet/gradle/play/internal/Progress.kt]: Can do!

**Pull request merged**